### PR TITLE
[device/dell] Added dynamic sai.profile generation

### PR DESCRIPTION
**Pull request by vharish02**
This commit addds new code for generating dynamic sai.profile file.
The sai.profile.j2 will generate the sai.profile dynamically based on
the topology. It will generate the sai.profile under /etc/sai.d/ directory
in syncd. Before syncd is started this J2 file will be run from the start.sh
file from /usr/bin/ directory. Since the sai.profile is dynamically generated
the old sai.profile file is not required so deleted the file for S6100.
It also address couple of more changes for port_config.ini support is added
for speed in the file which can be used later to find the port speed. Also
the buffer_default_t*.j2 file the ports should be from 0 to 64 changed.

Unit tested the code on S6100 for dynamic generation of sai.profile file for
both T0 and T1 and the file was created in /etc/sai.d/sai.profile
Similarly tested the sonic binary on S6000 to make sure that the sai.profile is
copied from the /usr/share/sonic/hwsku/sai.profile to /etc/sai.d/sai.profile.

Signed-off-by: Harish Venkatraman <Harish_Venkatraman@dell.com>

**- What I did**

**- How I did it**

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by vharish02** [comment]
@yxieca Can you please review the changes.

**Comment by yxieca** [comment]
You need to make swss change first, and add submodule update to this review.

File: platform/broadcom/docker-syncd-brcm/start.sh [hunk 1]
```diff
@@ -1,11 +1,24 @@
 #!/usr/bin/env bash
 
 PLATFORM_DIR=/usr/share/sonic/platform
+HWSKU_DIR=/usr/share/sonic/hwsku
 
 rm -f /var/run/rsyslogd.pid
 
+
```
Comment (yxieca) [inline platform/broadcom/docker-syncd-brcm/start.sh]: Can you remove this empty line?

File: device/dell/x86_64-dell_s6100_c2538-r0/Force10-S6100/sai.profile.j2 [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+{%-     if switch_role.lower() == 'torrouter' %}
+{%         set sai_profile_filename = 'SAI_INIT_CONFIG_FILE=/etc/bcm/th-s6100-64x40G-t0.config.bcm' -%}
+{%-     else %}
+{%-         set sai_profile_filename = 'SAI_INIT_CONFIG_FILE=/etc/bcm/th-s6100-64x40G-t1.config.bcm' -%}
+{%-     endif %}
+{%- else %}
+{%-     set sai_profile_filename = 'SAI_INIT_CONFIG_FILE=/etc/bcm/th-s6100-64x40G-t1.config.bcm' -%}
+{%- endif %}
+{# Write the contents of sai_ profile_filename to sai.profile file #}
+{{ sai_profile_filename }}
```
Comment (yxieca) [inline device/dell/x86_64-dell_s6100_c2538-r0/Force10-S6100/sai.profile.j2]: s/sai_profile_filename/sai_profile_contents/

**Comment by vharish02** [comment]
@yxieca All minor changes done.

**Pull request merged**